micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/TimeWindowFixedBoundaryHistogram.java

```diff
@@ -38,28 +38,65 @@ public class TimeWindowFixedBoundaryHistogram extends AbstractTimeWindowHistogra
 
     private final boolean isCumulativeBucketCounts;
 
+    /**
+     * Create a {@code TimeWindowFixedBoundaryHistogram} instance with cumulative bucket
+     * counts and buckets based on the {@link DistributionStatisticConfig config} and
+     * {@code supportsAggregablePercentiles}.
+     * @param clock clock
+     * @param config distribution statistic configuration
+     * @param supportsAggregablePercentiles whether the backend supports aggregable
+     * percentiles
+     * @see PercentileHistogramBuckets
+     */
     public TimeWindowFixedBoundaryHistogram(Clock clock, DistributionStatisticConfig config,
             boolean supportsAggregablePercentiles) {
         this(clock, config, supportsAggregablePercentiles, true);
     }
 
     /**
-     * Create a {@code TimeWindowFixedBoundaryHistogram} instance.
+     * Create a {@code TimeWindowFixedBoundaryHistogram} instance with buckets based on
+     * the {@link DistributionStatisticConfig config} and
+     * {@code supportsAggregablePercentiles}.
      * @param clock clock
      * @param config distribution statistic configuration
-     * @param supportsAggregablePercentiles whether it supports aggregable percentiles
-     * @param isCumulativeBucketCounts whether it uses cumulative bucket counts
+     * @param supportsAggregablePercentiles whether the backend supports aggregable
+     * percentiles
+     * @param isCumulativeBucketCounts whether to use cumulative bucket counts
      * @since 1.9.0
+     * @see PercentileHistogramBuckets
      */
     public TimeWindowFixedBoundaryHistogram(Clock clock, DistributionStatisticConfig config,
             boolean supportsAggregablePercentiles, boolean isCumulativeBucketCounts) {
+        this(clock, config, supportsAggregablePercentiles, isCumulativeBucketCounts, false);
+    }
+
+    /**
+     * Create a {@code TimeWindowFixedBoundaryHistogram} instance with buckets based on
+     * the {@link DistributionStatisticConfig config} and
+     * {@code supportsAggregablePercentiles}. This constructor allows for use cases that
+     * always need an infinity bucket in the histogram.
+     * @param clock clock
+     * @param config distribution statistic configuration
+     * @param supportsAggregablePercentiles whether the backend supports aggregable
+     * percentiles
+     * @param isCumulativeBucketCounts whether to use cumulative bucket counts
+     * @param includeInfinityBucket whether to always include an infinity bucket
+     * @since 1.13.11
+     * @see PercentileHistogramBuckets
+     */
+    protected TimeWindowFixedBoundaryHistogram(Clock clock, DistributionStatisticConfig config,
+            boolean supportsAggregablePercentiles, boolean isCumulativeBucketCounts, boolean includeInfinityBucket) {
```

Comment:
This `protected` constructor is the only (semi-)public API added/changed as part of this pull request. I tried to keep API changes minimal since this is in a maintenance branch. I would have liked to not add any, but I did not come up with a way to achieve that without doing something like copying everything from `TimeWindowFixedBoundaryHistogram` to an implementation of Histogram in the stackdriver module. That would probably be more prone to code rot and bugs.

Reply:
It is a wider API change but having an `includeInfinityBucket()` config method on `DistributionStatisticConfig` could spare the extra ctors in histograms.

Reply:
That is an interesting idea, but I'd rather avoid it in a maintenance release because the current change is more internal and if we are going to move toward making including the infinity bucket the only option on main, we'd be immediately deprecating the new public API.